### PR TITLE
[RLlib][Data][build_base] Fix RLlib test build rules that should trigger when there are dataset changes. 

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kouroshHakha** [comment]
nice: 
1/
image
2/ 
image
3/
image

**Review by bveeramani — APPROVED** [review]
Stamp

File: python/ray/data/dataset.py [hunk 1]
```diff
@@ -209,7 +209,7 @@ class Dataset(Generic[T]):
         >>> ds.map_batches(lambda batch: [v * 2 for v in batch])
         MapBatches(<lambda>)
         +- Dataset(num_blocks=17, num_rows=1000, schema=<class 'int'>)
-        >>> # Compute max.
+        >>> # Compute maximum
```
Comment (c21) [inline python/ray/data/dataset.py:212]: super nit: let's keep `.`, thanks.

**Review by c21 — APPROVED** [review]

File: python/ray/data/dataset.py (hunk 1 above)
Comment (kouroshHakha) [reply python/ray/data/dataset.py:212]: Can we merge this as is? I don't want to wait for the tests to pass again. This is kind of blocking visibility into RLlib ci failures. Alternatively I can add the dot but don't wait for the tests to pass.

Comment (c21) [reply python/ray/data/dataset.py:212]: Sure, go ahead.

**Review by gjoliver — APPROVED** [review]

**Pull request merged**